buffer/src/main/java/io/netty5/buffer/api/BufferAllocator.java

```diff
@@ -125,6 +126,49 @@ static BufferAllocator offHeapPooled() {
      */
     Buffer allocate(int size);
 
+    /**
+     * Compose the send of a buffer and present them as a single buffer.
+     * <p>
+     * When a composite buffer is closed, all of its constituent component buffers are closed as well.
+     * <p>
+     * See the class documentation for more information on what is required of the given buffers for composition to be
+     * allowed.
+     *
+     * @param send The sent buffer to compose into a single buffer view.
+     * @return A buffer composed of, and backed by, the given buffers.
+     * @throws IllegalStateException if one of the sends have already been received. The remaining buffers and sends
+     * will be closed and discarded, respectively.
+     */
+    default CompositeBuffer compose(Send<Buffer> send) {
```

Comment:
I was wondering if we should add also `default CompositeBuffer compose(Send<Buffer> first, Send<Buffer> second)` to reduce a bit of boiler-plate code. WDYT ?

Reply:
Why not use varargs like in the old API?

Reply:
I'm also wondering about this change.

Reply:
@chrisvest because you cant use `@SafeArgs` on interface methods :/